lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,394 @@
+
+    obs_index_header = hdu_index_header.copy()
+    obs_index_header["HDUCLAS2"] = "OBS"
+    obs_index_header["MJDREFI"] = event_table.meta["MJDREFI"]
+    obs_index_header["MJDREFF"] = event_table.meta["MJDREFF"]
+
+    obs_index = fits.BinTableHDU(
+        obs_index_table, header=obs_index_header, name="OBS INDEX"
+    )
+    obs_index_list = fits.HDUList([fits.PrimaryHDU(), obs_index])
+
+    return hdu_index_list, obs_index_list
+
+
+def create_event_list(data, run_number, source_name, effective_time, elapsed_time):
+    """
+    Create the event_list BinTableHDUs from the given data
+
+    Parameters
+    ----------
+        Data: DL2 data file
+                'astropy.table.QTable'
+        Run: Run number
+                Int
+        Source_name: Name of the source
+                Str
+        Effective_time: Effective time of triggered events of the run
+                Float
+        Elapsed_time: Total elapsed time of triggered events of the run
+                Float
+    Returns
+    -------
+        Events HDU:  `astropy.io.fits.BinTableHDU`
+        GTI HDU:  `astropy.io.fits.BinTableHDU`
+        Pointing HDU:  `astropy.io.fits.BinTableHDU`
+    """
+
+    tel_list = np.unique(data["tel_id"])
+
+    # Timing parameters
+    t_start = data["dragon_time"].value[0]
+    t_stop = data["dragon_time"].value[-1]
+    time = Time(data["dragon_time"], format="unix", scale="utc")
+    date_obs = time[0].to_value("iso", "date")
+
+    # Position parameters
+    reco_alt = data["reco_alt"]
+    reco_az = data["reco_az"]
+    pointing_alt = data["pointing_alt"]
+    pointing_az = data["pointing_az"]
+
+    src_sky_pos = SkyCoord(
+        alt=reco_alt, az=reco_az, frame=AltAz(obstime=time, location=location)
+    ).transform_to(frame="icrs")
+    tel_pnt_sky_pos = SkyCoord(
+        alt=pointing_alt[0],
+        az=pointing_az[0],
+        frame=AltAz(obstime=time[0], location=location),
+    ).transform_to(frame="icrs")
+
+    try:
+        object_radec = SkyCoord.from_name(source_name)
+    except Exception:
+        log.error("Name resolve error in finding the Object in Sesame")
+        object_radec = SkyCoord(tel_pnt_sky_pos.icrs)
+
+    # Observation modes
+    source_pointing_diff = object_radec.separation(
+        SkyCoord(tel_pnt_sky_pos.ra, tel_pnt_sky_pos.dec)
+    ).deg
+
+    if round(source_pointing_diff, 1) == wobble_offset:
+        mode = "WOBBLE"
+    elif round(source_pointing_diff, 1) > 1:
+        mode = "OFF"
+    elif round(source_pointing_diff, 1) == 0.0:
+        mode = "ON"
+    else:
+        # Nomenclature is to be worked out or have a separate way to mark mispointings
+        mode = "UNDETERMINED"
+
+    log.error(
+        f"Source pointing difference with camera pointing is {source_pointing_diff:.3f} deg"
+    )
+
+    event_table = QTable(
+        {
+            "EVENT_ID": u.Quantity(data["event_id"]),
+            "TIME": u.Quantity(data["dragon_time"]),
```

Comment:
Missing unit

Reply:
`dragon_time` already has the units when I read it into QTable format

Reply:
Then why call `u.Quantity` again?

Reply:
Yes, I am removing it